### PR TITLE
Enhance angular 1.5 types

### PR DESCRIPTION
**Pull request by villesau**
Slightly enhanced angular 1.5 types. E.g fixed forEach, and added $stateProvider.

### CODE REVIEW

File: definitions/npm/angular_v1.5.x/flow_v0.47.x-/angular_v1.5.x.js [hunk 1]
```diff
@@ -91,7 +89,6 @@ declare module angular {
   ) => AngularModule
 
   declare type ConfigDeclaration = (
-    name: string,
```
Comment (villesau) [inline definitions/npm/angular_v1.5.x/flow_v0.47.x-/angular_v1.5.x.js]: this was wrong. Config does not get name.

File: definitions/npm/angular_v1.5.x/flow_v0.47.x-/test_angular_v1.5.x.js [hunk 2]
```diff
@@ -64,11 +56,11 @@ describe("directives", () => {
   });
 
   it("requires proper restrict when defined", () => {
+    // $ExpectError
```
Comment (villesau) [inline definitions/npm/angular_v1.5.x/flow_v0.47.x-/test_angular_v1.5.x.js]: error location has changed in some version

**Comment by villesau** [comment]
The errors comes from 1.4.7 version of angular, see related issue: https://github.com/flow-typed/flow-typed/issues/2752

**Comment by AndrewSouthpaw** [comment]
Still has failing Travis tests @villesau

**Comment by villesau** [comment]
@AndrewSouthpaw it's different libdefs which fails.

**Comment by GAntoine** [comment]
Merging in since the failing specs are unrelated and andrew gave the all clear.

**Pull request merged**